tests/test_controller.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+from deepdiff import DeepDiff
+import tables
+
+import analysis.analysisfunctions as af
+from test_python_worker import pythonworker_enviroment
+import controller
+from hdf5logger import hdf5collector
+
+
+def compare_hdf5_files():
```

Comment:
Locally, I used `h5diff` two compare an output vs the golden hdf5 file.

Can we also use this here (or a similar tool)?

Reply:
I have no experience with h5diff. What i want to avoid is a failed test if the experiment index numbers change. Can h5diff cope with it?

Reply:
It will most likely print an error as the files don't match. Does this happen as well with only a single runner? I would guess that the output should be deterministic.

Reply:
I will not use h5diff, as it is not python native and i want to use python assert. However i will use the same trick i am using to compress the collected analysis data from the experiment datas and check if it is empty.

Reply:
Now the testbenches also assert if indexing stays consistant by asserting the experiment group title and testing extracted index numbers.